### ISSUE TITLE
Auth token with Null username and password

### ISSUE THREAD

**Issue by GabrieleBazzottiPers** [openwrt/luci#1700]
When I try to log in with
`curl https://<hostname>/cgi-bin/luci/rpc/auth --data '{ "id":1, "method":"login", "params" : [null,null] }'`
I get `{"id":1,"result":"a3aa53c842f6ec306e1480a4c7cef5b1","error":null}`
The token don't let me do command like get_all and set but it should't  exist in the response.

**Comment by sophana** [comment]
in sys.lua, user.checkpasswd looks really suspicious. It returns true when pwh == nil???
A simple fix would be to check username, but I think there is a big (and severe) problem there.
```
function user.checkpasswd(username, pass)
	local pwh, pwe = user.getpasswd(username)
	if pwe then
		return (pwh == nil or nixio.crypt(pass, pwh) == pwh)
	end
	return false
end
```

**Comment by jow-** [comment]
> in sys.lua, user.checkpasswd looks really suspicious. It returns true when pwh == nil???

When pwh is nil, no password is set in `/etc/shadow`, in this case access is allowed regardless of the provided password.

**Comment by jow-** [comment]
Fixed with feefc60 - `null, null` authentication now returns `null` instead of a defunct token
